src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -1292,4 +1293,1368 @@ void EIRPlantLoopHeatPump::oneTimeInit(EnergyPlusData &state)
+                    sourceSideOutletNodeName += ("_" + std::to_string(rand()));
+                }
+
+                // A5
+                if (thisPLHP.EIRHPType == DataPlant::PlantEquipmentType::HeatPumpFuelFiredHeating) {
+                    if (fields.find("companion_cooling_heat_pump_name") != fields.end()) { // optional field
+                        thisPLHP.companionCoilName = UtilityRoutines::MakeUPPERCase(fields.at("companion_cooling_heat_pump_name").get<std::string>());
+                    }
+                } else if (thisPLHP.EIRHPType == DataPlant::PlantEquipmentType::HeatPumpFuelFiredCooling) {
+                    if (fields.find("companion_heating_heat_pump_name") != fields.end()) { // optional field
+                        thisPLHP.companionCoilName = UtilityRoutines::MakeUPPERCase(fields.at("companion_heating_heat_pump_name").get<std::string>());
+                    }
+                }
+
+                // A6 Fuel Type
+                std::string tempRsrStr = UtilityRoutines::MakeUPPERCase(fields.at("fuel_type").get<std::string>());
+                thisPLHP.fuelType = DataGlobalConstants::AssignResourceTypeNum(tempRsrStr);
+                // Validate fuel type input
+                // Locals
+                static constexpr std::string_view RoutineName("processInputForEIRPLHP: ");
+                bool FuelTypeError(false);
+                UtilityRoutines::ValidateFuelTypeWithAssignResourceTypeNum(tempRsrStr, tempRsrStr, thisPLHP.fuelType, FuelTypeError);
+                if (FuelTypeError) {
+                    ShowSevereError(state, format("{}{}=\"{}\",", RoutineName, cCurrentModuleObject, thisPLHP.name));
+                    // ShowContinueError(state, "Invalid " + state.dataIPShortCut->cAlphaFieldNames(2) + '=' + state.dataIPShortCut->cAlphaArgs(2));
+                    ShowContinueError(state, format("Invalid Fuel Type = {}", tempRsrStr));
+                    // Set to Electric to avoid errors when setting up output variables
+                    tempRsrStr = "NaturalGas";
+                    thisPLHP.fuelType = DataGlobalConstants::ResourceType::Natural_Gas;
+                    errorsFound = true;
+                    FuelTypeError = false;
+                }
+
+                // A7 End use category
+                thisPLHP.endUseSubcat = UtilityRoutines::MakeUPPERCase(fields.at("end_use_subcategory").get<std::string>());
+                // 2022-05-13: default: empty?
+                if (thisPLHP.endUseSubcat == "") {
+                    thisPLHP.endUseSubcat = "Heat Pump Fuel Fired"; // or "General"?
+                }
+
+                // N1 Nominal heating capacity
+                auto tmpRefCapacity =
+                    fields.at((thisPLHP.EIRHPType == DataPlant::PlantEquipmentType::HeatPumpFuelFiredHeating) ? "nominal_heating_capacity"
+                                                                                                              : "nominal_cooling_capacity");
```

Comment:
This line is a bit difficult to read.  I think it's worth a temporary variable to store the result of the ternary operation:

```
std::string const capacityFlag = (thisPLHP.EIRHPType == DataPlant::PlantEquipmentType::HeatPumpFuelFiredHeating) ? "nominal_heating_capacity" : "nominal_cooling_capacity"
auto tmpRefCapacity = fields.at(capacityFlag);
```

A bonus would be to declare the string above when you are assigning the value of `thisPLHP.EIRHPType` itself, since you are already inside an IF block up there.  Then you don't have to have a second IF here (ternary).

Reply:
For this one: https://github.com/NREL/EnergyPlus/pull/9405#discussion_r1106317643

I tried to remove the two lines but it turned out that the number of object check is needed. Probably due to for loop elments of:
`    for (auto &classToInput : classesToInput) {`
There are always two elements in `classesToInput`---one for the `Cooling` coil and one for the `Heating` coil. Without the number of object check, it would fail if in a configuration that only one type of object (either Cooling or Heating, which is perfectly valid) is specified in the input file---actually this is the case in the unit test file.

So finally I still added the number of object check, although removed the local variable `numPLHP`.

Reply:
The few comments above were all addressed in previous changes and discussions, including the "bonus" one (https://github.com/NREL/EnergyPlus/pull/9405#discussion_r1106321490).